### DISCUSSION TITLE
Stripe payment integration

### DISCUSSION THREAD

**Discussion by Super0109** [community/community#60996] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

Hello, everyone,

Now, I am doing the Stripe payment integration on the shopping site.
My shopping site made by React,js for Frontend.
But I faced on one thing.
How can I catch event when the stripe payment successed?

**Comment by SuperSmile0109** [comment, marked as answer]
Hi,  please try this.

    stripe.retrievePaymentIntent(clientSecret).then(({ paymentIntent }) => {
      console.log(paymentIntent);
      switch (paymentIntent.status) {
        case "succeeded":
          toast.success("Payment succeeded!");
          break;
        case "processing":
          toast.info("Your payment is processing.");
          break;
        case "requires_payment_method":
          toast.error("Your payment was not successful, please try again.");
          break;
        default:
          toast.error("Something went wrong.");
          break;
      }
    });
  }, [stripe]);

**Reply by Super0109** [reply]
Thank you so much.

Working well.

**Reply by Brandonb0981** [reply]
I need to connect a payment method to my shopify online store

**Reply by Brandonb0981** [reply]
Can you help me

**Reply by SuperSmile0109** [reply]
Sure, I can help you.

How can I contact with you?

Thanks

**Reply by SuperSmile0109** [reply]
Please reach to me when you are free. 

Let me share my contact info:
Skype: live:.cid.f0d117f6fd70c986
Discord: james9381
Telegram: JamesDev109

**Reply by Brandonb0981** [reply]
I need to add payment method to Spotify store
